docs/types/reference-types.rst

```diff
@@ -217,6 +221,10 @@ the first element to ``uint``.
         }
     }
 
```

Comment:
```suggestion
The type of ``[0, 50, 257, 1]`` is ``uint16[4]`` because that's the type of literal ``257`` and the first type on the list that all the values can be implicitly converted to.
```

I'd also be tempted to add this, because it follows the rules stated above, but I'm not sure if it's a good idea if it won't compile on 0.8.0:
```suggestion
The type of ``[[0x1, 1], [0xffffff, 2], [0xff, 3], [0xffff, 4]]`` is ``uint24[2][4]`` because there's an implicit conversion possible from ``uint8[2]`` and ``uint16[2]`` to ``uint24[2]``.
```